src/compiler/parser.ts

```diff
@@ -6655,8 +6692,58 @@ namespace ts {
                     }
                 }
 
+                function escapedTextsEqual(a: EntityName, b: EntityName): boolean {
+                    while (!ts.isIdentifier(a) || !ts.isIdentifier(b)) {
+                        if (!ts.isIdentifier(a) && !ts.isIdentifier(b) && a.right.escapedText === b.right.escapedText) {
+                            a = a.left;
+                            b = b.left;
+                        }
+                        else {
+                            return false;
+                        }
+                    }
+                    return a.escapedText === b.escapedText;
+                }
+
+                function parseChildParameterOrPropertyTag(target: PropertyLikeParse.Property): JSDocTypeTag | JSDocPropertyTag | false;
+                function parseChildParameterOrPropertyTag(target: PropertyLikeParse.Parameter, name: EntityName): JSDocParameterTag | false;
+                function parseChildParameterOrPropertyTag(target: PropertyLikeParse, name?: EntityName): JSDocTypeTag | JSDocPropertyTag | JSDocParameterTag | false {
+                    let canParseTag = true;
+                    let seenAsterisk = false;
+                    while (true) {
+                        nextJSDocToken();
+                        switch (token()) {

```

Comment:
Not really; we hit the default case a lot for punctuation and identifiers that we want to skip as part of the tag's description. 

Note that it's the end of the comment, not the end of the whole file. And we return false as soon as `tryParseChildTag` encounters any tag that is not `param`, `type` or `property`. So runaway parsing shouldn't be a big problem.

Also note: Comments aren't handled by this PR but once they are, `default` will probably look like `parseJSDocCommentWorker`, where it just pushes the token into a comment array.